lib/class-wp-rest-block-editor-settings-controller.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * REST API: WP_REST_Block_Editor_Settings_Controller class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to retrieve the block editor settings via the REST API.
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Block_Editor_Settings_Controller extends WP_REST_Controller {
+	/**
+	 * Constructs the controller.
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental/wp-block-editor/v1';
+		$this->rest_base = 'settings';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read block editor settings
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has permission, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
+
+		return new WP_Error(
+			'rest_cannot_read_block_editor_settings',
+			__( 'Sorry, you are not allowed to read the block editor settings.', 'gutenberg' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Returns the block editor's settings
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$context  = ! empty( $request['context'] ) ? $request['context'] : 'post-editor';
+		$settings = array_merge(
+			apply_filters( 'block_editor_settings', array() ),
+			gutenberg_get_block_editor_settings( $context )
```

Comment:
`gutenberg_get_block_editor_settings` has `block_editor_setting` filter internally.

Reply:
Thanks! That solves the issue I mentioned [here](https://github.com/WordPress/gutenberg/pull/29969#discussion_r616708627) 😅  

I still had it because I wanted to have the global styles settings but I'll wait until this [work](https://github.com/WordPress/gutenberg/issues/30082#issuecomment-816679803) lands. So I'll remove that 👍

Reply:
If a plugin defines a filter that uses `$post` object in the callback passed to `block_editor_settings` then it will throw an error. 

It's all covered in `gutenberg_get_block_editor_settings`:

https://github.com/WordPress/gutenberg/blob/a88a3b587cf203c62c196332d64b33afe164d283/lib/block-editor.php#L277-L290